zoneminder/zm.py

```diff
@@ -72,12 +92,17 @@ def change_state(self, api_url, post_data) -> dict:
     def _zm_request(self, method, api_url, data=None,
                     timeout=DEFAULT_TIMEOUT) -> dict:
         """Perform a request to the ZoneMinder API."""
+        token_url_suffix = ""
+        if self._auth_token:
+            token_url_suffix = "?token=" + self._auth_token
         try:
             # Since the API uses sessions that expire, sometimes we need to
             # re-auth if the call fails.
             for _ in range(ZoneMinder.LOGIN_RETRIES):
                 req = requests.request(
-                    method, urljoin(self._server_url, api_url), data=data,
+                    method,
+                    urljoin(self._server_url, api_url) + token_url_suffix,
```

Comment:
Can we do this with a `urljoin` rather than string concat?

Reply:
`urljoin` can't join query parameters, the second argument would just override the arguments passed as the base. What problem are you trying to solve?

Reply:
@rohankapoorcom ^

Reply:
Sorry - I was coming at this from a Java standpoint.... I was trying to avoid the string concats.